### ISSUE TITLE
v5 Install Fails on MW 1.29

### ISSUE THREAD

**Issue by QuotesUK** [JeroenDeDauw/Maps#414]
### Setup

MediaWiki | 1.29.2
PHP | 7.0.29 (cgi-fcgi)
MySQL | 5.1.73
ICU | 4.2.1

### Issue

With a clean installation of MW 1.29 I used composer to install Maps. I used the JSON example given in the instructions as "mediawiki/maps": "~5.0"

Package operations: 12 installs, 0 updates, 42 removals

The result is a fatal error.

Fatal error: Uncaught Error: Class 'Wikimedia\Assert\Assert' not found

**Comment by JeroenDeDauw** [comment]
You need to add `mediawiki/maps` to the composer file and not remove what is already there.

For more info see https://github.com/JeroenDeDauw/Maps/blob/master/INSTALL.md#maps-installation

**Comment by QuotesUK** [comment]
MW 1.29.2 is clean installed by file upload, then continuation of installation process to generate LocalSettings.php. Then composer.phar is uploaded with a composer.json file as per the instructions you linked to:

    "require": {
        "mediawiki/maps": "~5.0"
    }

There is no "already there". First run for composer is as above.

**Comment by kghbln** [comment]
There may also be an issue with Composer itself. The current version 1.6.4 is broken. Try to run composer more than one time. Moreover use "composer.local.json" as preferred suggestion.

**Comment by kghbln** [comment]
I assume that things worked out fine in the meantime, meaning that the issue is resolved. Thus closing. If the issue persists please re-open with additional information.

**Comment by QuotesUK** [comment]
Apologies for not updating sooner, I have only just found my way forward.

Thank you for advising me to get composer sorted first - that worked.

From a clean installation I ran composer on its own. That gave me a list of commands so I ran test mode and found an error, lint was missing, so I fixed that first. However instead of adding a composer.json file I just used the cron job to execute the installation.

/path/to/local/server/httpdocs/composer.phar

--working-dir=/path/to/local/server/httpdocs/ require --dev jakub-onderka/php-parallel-lint

--working-dir=/path/to/local/server/httpdocs/ require --dev mediawiki/maps --prefer-stable

After adding my Google API key and updating local settings the extension is working.
